test/new-e2e/tests/installer/windows/suites/apm-library-dotnet-package/msi_install_test.go

```diff
@@ -0,0 +1,363 @@
+}
+
+// TestMSISkipRollbackIfInstalled tests that the newly installed dotnet library is not removed when the MSI upgrade fails
+// if another version of the library was already installed.
+func (s *testAgentMSIInstallsDotnetLibrary) TestMSISkipRollbackIfInstalled() {
+	// Arrange
+	s.setAgentConfig()
+	oldVersion := s.previousDotnetLibraryVersion
+	newVersion := s.currentDotnetLibraryVersion
+
+	// Install first version
+	s.installPreviousAgentVersion(
+		installerwindows.WithMSIArg("DD_APM_INSTRUMENTATION_ENABLED=iis"),
+		// TODO: remove override once image is published in prod
+		// TODO: support DD_INSTALLER_REGISTRY_URL
+		installerwindows.WithMSIArg("SITE=datad0g.com"),
+		installerwindows.WithMSIArg(fmt.Sprintf("DD_APM_INSTRUMENTATION_LIBRARIES=dotnet:%s", oldVersion.PackageVersion())),
+		installerwindows.WithMSILogFile("install.log"),
+	)
+
+	// Rollback during upgrade to the new version
+	err := s.Installer().Install(
+		installerwindows.WithMSIArg("DD_APM_INSTRUMENTATION_ENABLED=iis"),
+		// TODO: remove override once image is published in prod
+		// TODO: support DD_INSTALLER_REGISTRY_URL
+		installerwindows.WithMSIArg("SITE=datad0g.com"),
+		installerwindows.WithMSIArg(fmt.Sprintf("DD_APM_INSTRUMENTATION_LIBRARIES=dotnet:%s", newVersion.PackageVersion())),
+		installerwindows.WithMSILogFile("install-rollback.log"),
+		installerwindows.WithMSIArg("WIXFAILWHENDEFERRED=1"),
+		installerwindows.WithMSIDevEnvOverrides("CURRENT_AGENT"),
+	)
+	s.Require().Error(err)
+
+	// Start IIS and check that the NEW version of the library is loaded
+	defer s.stopIISApp()
+	s.startIISApp(webConfigFile, aspxFile)
+	newLibraryPath := s.getLibraryPathFromInstrumentedIIS()
+	s.Require().Contains(newLibraryPath, newVersion.Version())
+}
+
+// TestUninstallKeepsLibrary tests that the dotnet library is not removed when the Agent MSI is uninstalled
+func (s *testAgentMSIInstallsDotnetLibrary) TestUninstallKeepsLibrary() {
+	s.setAgentConfig()
+	version := s.currentDotnetLibraryVersion
+
+	// Install the dotnet library with the Agent MSI
+	s.installCurrentAgentVersion(
+		installerwindows.WithMSIArg("DD_APM_INSTRUMENTATION_ENABLED=iis"),
+		// TODO: remove override once image is published in prod
+		installerwindows.WithMSIArg("SITE=datad0g.com"),
+		installerwindows.WithMSIArg(fmt.Sprintf("DD_APM_INSTRUMENTATION_LIBRARIES=dotnet:%s", version.PackageVersion())),
+		installerwindows.WithMSILogFile("install.log"),
+	)
+
+	// Uninstall the Agent
+	err := s.Installer().Uninstall(
+		installerwindows.WithMSILogFile("uninstall.log"),
+	)
+	s.Require().NoError(err)
+
+	// Start the IIS app to load the library
+	defer s.stopIISApp()
+	s.startIISApp(webConfigFile, aspxFile)
+
+	// Check that the expected version of the library is loaded
+	oldLibraryPath := s.getLibraryPathFromInstrumentedIIS()
+	s.Require().Contains(oldLibraryPath, version.Version())
+}
+
+func (s *testAgentMSIInstallsDotnetLibrary) setAgentConfig() {
```

Comment:
Why is this needed?

Reply:
The tests use the daemon to perform remote upgrades, so we have to enable the daemon in the config.